typed-builder-macro/src/struct_info.rs

```diff
@@ -85,15 +87,27 @@ impl<'a> StructInfo<'a> {
                 empty_type()
             }
         }));
-        let init_fields_expr = self.included_fields().map(|f| {
-            f.builder_attr.via_mutators.as_ref().map_or_else(
-                || quote!(()),
-                |via_mutators| {
-                    let init = &via_mutators.init;
-                    quote!((#init,))
-                },
-            )
-        });
+        let builder_method_const = Rc::new(OnceCell::new());
+        let init_fields_expr = self
+            .included_fields()
+            .map({
+                let builder_method_const = builder_method_const.clone();
+                move |f| {
+                    f.builder_attr.via_mutators.as_ref().map_or_else(
+                        || quote!(()),
+                        |via_mutators| {
+                            let init = &via_mutators.init;
+                            if !matches!(init, syn::Expr::Lit(_)) {
```

Comment:
I like to make things explicit. If the build method is to be `const`, it should be marked as such in the configuration attributes - not implicitly deduced based on some incomplete (literals aren't the only `const` expressions) heuristics.

Reply:
True, makes sense. It was just my compulsive thought to make it const when I saw the macro expansion for my usecase. But now I know that it is not so simple